python_modules/libraries/dagster-airbyte/dagster_airbyte/dagster_airbyte_translator.py

```diff
@@ -0,0 +1,67 @@
+from dagster import AssetKey
+
+from dagster_airbyte.asset_defs import AirbyteConnectionMetadata
+
+
+class DagsterAirbyteTranslator:
```

Comment:
Peanut gallery checking in, but if we're making a concerted effort here to improve our suite of integrations, I think now would be an excellent time to do a bit of cross-integration refactoring / abstraction.

The most glaring thing to me is that `DagsterAirbyteTranslator` and `DagsterDbtTranslator` should really share a common base class (and probably "think" in terms of AssetSpecs, even if we retain the exploded methods for backcompat).

i.e. there should be a generic:

```python
class DagsterTranslator(Generic[T]):
    
    def get_asset_spec(identifier: str, metadata: T) -> AssetSpec:
        ...
```

or something of the sort, which all of our integrations can inherit from. Then `DagsterAirbyteTranslator` would look like:

```python
class DagsterAirbyteTranslator(DagsterTranslator[AirbyteConnectionMetadata]):
    ...
```

and users could override things like:

```python
class MyCustomTranslator(DagsterAirbyteTranslator):

    def get_asset_spec(identifier: str, metadata: AirbyteConnectionMetadata) -> AssetSpec:
         return super().get_asset_spec(identifier, metadata)._replace(
              <stuff i want to replace>
         )
```

This a) reduces boilerplate and b) means that we don't need to update every single one of these things if we ever decide to add something to AssetSpec. I also thing it's probably generally more convenient for a user to write a single function that sets multiple properties of a spec rather than one function per property.

So that's the Translator side of the world, but I also think there's likely an opportunity to make a generic "integration assets decorator", or at least simplify the process a bunch -- assuming we get to the point where we can just make a simple function that returns a list of AssetSpecs (with the help of this Translator object), then we can probably create some generic function which takes in:

- domain specific function to get some raw blob of metadata (e.g. for dbt this would be the code to get the manifest.json file)
- a translator object to turn that into specs

and then combines those together to create a decorator. this bit is probably less pressing than the Translator standardization piece, mostly just tossing it out there as an idea.

TL;DR: I really think we should standardize these things sooner rather than later to make future integration work faster / easier / more consistent